__init__.py

```diff
@@ -1150,6 +1151,38 @@ def _speak_weather(self, dialog):
         self.log.info("Speaking dialog: " + dialog.name)
         self.speak_dialog(dialog.name, dialog.data, wait=True)
 
+    @skill_api_method
+    def get_current_weather(self, message=None):
+        """Get the current temperature and condition code.
+        
+        Args:
+            message (Message): [Optional] mock message will be parsed to
+                               determine weather intent.
+
+        Returns:
+            Dict: {
+                temperature: current temperature
+                high_temperature: forecasted high for today
+                low_temperature: forecasted low for today
+                condition_code: code representing overall weather condition
+                                see Maps for all codes in skill/weather.py
+            }
+        """
+        self.log.debug("Requested current weather information:")
+        if message is None:
+            message = Message("", data={"utterance": ""})
```

Comment:
maybe this would be better handled in the `self._get_intent_data` method? if there isn't an utterance there must be some checks that can be skipped and optimized further down

ie, make message optional in `self._get_intent_data`

Reply:
I propose an alternate approach...

```python
try:
    weather = self.weather_api.get_weather_for_coordinates(
        self.config_core.get("system_unit"), 
        self.weather_config.latitude, 
        self.weather_config.longitude
    )
except Exception:
    self.log.exception("Unexpected error getting weather for skill API.")
```

The weather API only needs data from the WeatherConfig object, so you don't actually need the intent object.  Also, if you call `self._get_weather()` and an error occurs, the device will start speaking without being prompted, which would not be appropriate behavior for an API call, IMO.

Reply:
yeah much nicer - changed